### PR TITLE
Development section frontend

### PR DESCRIPTION
**Pull request by cp-ashish-k**
## Review checklist

**Note:** Make sure all points should be checked before submiting PR for review, otherwise PR will be cosidered invalid.

- [x] Follow guidelines for creating branch name, adding commit messages or PR title
- [x] Add MR description if applicable
- [x] Follow guidelines for naming conventions everywhere (i.e files/folders, data structures)
- [ ] Reuse code (if the same code is written twice, make it common and reuse it at both places)
- [ ] Remove unused or commented code if not required
- [ ] Make MR self-approved - review MR as a reviewer and give it self-approval if everything is ok, if not make the required changes
- [ ] In progress MR should be marked as draft
- [ ] Make MR mark as ready before submitting it for review
- [x] Add images, videos, or gifs of UI
- [x] Add page speed score screenshot of UI (It should be > 85 for mobile and > 90 for desktops)

- Page speed score screenshot of mobile UI
image

- Page speed score screenshot of desktop UI
image

- Mobile UI Video :
[Screencast from 28-02-24 10:47:47 AM IST.webm](https://github.com/canopas/canopas-website/assets/85920585/5a2b6de4-b192-4023-93be-d297f8810cec)

- Desktop UI Video :
[Screencast from 28-02-24 10:47:10 AM IST.webm](https://github.com/canopas/canopas-website/assets/85920585/c632d87f-dbd9-4a11-a8e2-2a246c3e410e)

### CODE REVIEW

File: nuxt-frontend/components/frontend-development/DevelopmentSection.vue [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  },
+];
+onMounted(() => {
+  window.addEventListener("scroll", handleScroll);
+});
+onUnmounted(() => {
+  window.removeEventListener("scroll", handleScroll);
+});
+function handleScroll() {
+  const child = document.querySelectorAll(".item");
```
Comment (cp-dharti-r) [inline nuxt-frontend/components/frontend-development/DevelopmentSection.vue:160]: use some meaningful name for classes that never duplicate

File: nuxt-frontend/components/frontend-development/DevelopmentSection.vue [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        <div class="swiper-pagination"></div>
+      </swiper>
+    </div>
+    <!-- Mobile Swiper UI ends-->
+    <!-- Desktop Swiper UI starts-->
+    <div class="pt-24 xl:pt-20 main hidden lg:block">
+      <div
+        v-for="(item, index) in items"
+        :key="index"
+        class="child-content bg-white pb-32 sticky top-96"
```
Comment (cp-jagruti-a) [inline nuxt-frontend/components/frontend-development/DevelopmentSection.vue:82]: The animation is not same as figma cards should will cross at some degree and opacity of that cards will change upto 50% refer figma properly

File: nuxt-frontend/components/frontend-development/DevelopmentSection.vue [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    <!-- Desktop Swiper UI starts-->
+    <div class="pt-24 xl:pt-20 main hidden lg:flex flex-col gap-32">
+      <div
+        v-for="(item, index) in items"
+        :key="index"
+        class="child-content bg-white sticky top-96 mb-52"
+        :class="[
+          item.classList,
+          index === activeIndex
+            ? `transition-all duration-100 -rotate-6 opacity-90 ease-in`
```
Comment (cp-jagruti-a) [inline nuxt-frontend/components/frontend-development/DevelopmentSection.vue:86]: According to figma we have to rotate cards in opposite direction if first is rotate left then second will rotate right like that

File: nuxt-frontend/components/frontend-development/DevelopmentSection.vue [hunk 4]
```diff
@@ -0,0 +1,194 @@
+<template>
+  <section
+    class="my-16 lg:mt-60 lg:mb-0 relative 3xl:h-[280vh] 3xl:outer-container"
+  >
+    <div
+      class="flex flex-col gap-4 lg:gap-6 container text-center lg:sticky top-28 3xl:!pb-0"
+      id="parentContainer"
```
Comment (cp-dharti-r) [inline nuxt-frontend/components/frontend-development/DevelopmentSection.vue:7]: let's use unique name for id

File: nuxt-frontend/components/frontend-development/DevelopmentSection.vue [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+              'bg-white-smoke': index % 2 !== 0,
+            }"
+            class="container py-6 rounded-xl"
+          >
+            <h3 class="mobile-header-3-semibold text-black-87">
+              {{ item.title }}
+            </h3>
+            <div class="sub-h3-regular text-black-60 mt-4">
+              <span
+                v-html="item.description"
```
Comment (cp-dharti-r) [inline nuxt-frontend/components/frontend-development/DevelopmentSection.vue:67]: `<span>` within `<span>` is not a good structure.
You can directly set the description into div, no need to use span for it.

File: nuxt-frontend/components/frontend-development/DevelopmentSection.vue [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+            'bg-white-smoke': index % 2 !== 0,
+          }"
+          class="p-10 container rounded-xl w-[45rem] shadow-lg h-[33rem]"
+        >
+          <h3 class="desk-header-3 text-black-87">
+            {{ index + 1 }}.
+            {{ item.title }}
+          </h3>
+          <div class="mobile-header-3-medium text-black-60 mt-4">
+            <span v-html="item.description" class="flex gap-4 flex-col"></span>
```
Comment (cp-dharti-r) [inline nuxt-frontend/components/frontend-development/DevelopmentSection.vue:97]: same here

**Review by cp-dharti-r — APPROVED** [review]

**Pull request merged**